Support attaching to an existing session

DirectXMan12 opened the pull request:
This commit adds support for attaching to an existing session.
To do so, use the '-a' or '--attach' flag with the session identifier
printed out in the gssapi-console banner (for example, the identifier
for the krb5 mech is the tmp directory path).

**NB**: Currently, this bundles a modified version of k5test.  Eventually, I would like to split it out from both this and python-gssapi entirely.

DirectXMan12 committed:
Support attaching to an existing session

This commit adds support for attaching to an existing session.
To do so, use the '-a' or '--attach' flag with the session identifier
printed out in the gssapi-console banner (for example, the identifier
for the krb5 mech is the tmp directory path).

frozencemetery commented:
This seems to work for me with python3 (python2 has other problems as discussed and so has not been tested), so merge when ready.

Also, :+1: for splitting out **k5test**.

DirectXMan12 commented:
Ok, fixed Python 2.7 issues.  I'll work on splitting out k5test.

DirectXMan12 committed:
Use standalone k5test

This commit switches from using the k5test bundled with python-gssapi
to using the standalone version of k5test.

DirectXMan12 commented:
This now is set up to use k5test, which has now been published as v0.9.0.
